### PR TITLE
Add static labels support for the AWS Firehose Logs

### PR DESCRIPTION
**Pull request by andriikushch**
#### PR Description

Adds support for setting some static labels at the AWS configuration level.

Static labels can be set with the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header.

Label names should be prefixed with `lbl_`. Label names and label values should be compatible with a [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) specification.

Example of the valid `X-Amz-Firehose-Common-Attributes` value with two custom labels:
```json
{
  "commonAttributes": {
    "lbl_label1": "value1",
    "lbl_label2": "value2"
  }
}
```

#### Which issue(s) this PR fixes

N.A.

#### Notes to the Reviewer
N.A.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/reference/components/loki.source.awsfirehose.md [hunk 1]
```diff
@@ -64,6 +64,19 @@ loki.source.awsfirehose "LABEL" {
 The component will start an HTTP server on the configured port and address with the following endpoints:
 
 - `/awsfirehose/api/v1/push` - accepting `POST` requests compatible with [AWS Firehose HTTP Specifications](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html).
+- `static_configs_labels` - query parameter can be used to set extra static labels. It should contain base64 encoded YAML with labels values. For example:
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki.source.awsfirehose.md:67]:
```suggestion

You can use the `static_configs_labels` query parameter to set extra static labels. It must contain base64 encoded YAML with labels values.
```

This isn't an endpoint... it's a query param. So it is probably better to add it as it's own sentence vs grouping it with the endpoint description.

File: docs/sources/reference/components/loki.source.awsfirehose.md [hunk 2]
```diff
@@ -64,6 +64,19 @@ loki.source.awsfirehose "LABEL" {
...
+
+    Generate base64 encoded string from the labels list:
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki.source.awsfirehose.md:69]:
```suggestion
    For example, you can generate a base64 encoded string from the labels list:
```

File: docs/sources/reference/components/loki.source.awsfirehose.md [hunk 3]
```diff
@@ -64,6 +64,19 @@ loki.source.awsfirehose "LABEL" {
...
+    ```bash
+    cat << EOF | base64
+    ---
+    label1: "value1"
+    label2: "value2"
+    EOF
+    LS0tCmxhYmVsMTogInZhbHVlMSIKbGFiZWwyOiAidmFsdWUyIgo=
+    ```
+  
+    Add query parameter to the url `/awsfirehose/api/v1/push?static_configs_labels=LS0tCmxhYmVsMTogInZhbHVlMSIKbGFiZWwyOiAidmFsdWUyIgo=`
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki.source.awsfirehose.md:79]:
```suggestion
    Add the query parameter with the base64 encoded value to the URL `/awsfirehose/api/v1/push?static_configs_labels=LS0tCmxhYmVsMTogInZhbHVlMSIKbGFiZWwyOiAidmFsdWUyIgo=`
```

**Comment by andriikushch** [comment]
Thank you, @clayton-cornell for the suggestions 👍

File: docs/sources/reference/components/loki.source.awsfirehose.md [hunk 4]
```diff
@@ -65,6 +65,19 @@ The component will start an HTTP server on the configured port and address with
 
 - `/awsfirehose/api/v1/push` - accepting `POST` requests compatible with [AWS Firehose HTTP Specifications](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html).
 
+You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels. Label names should be prefixed with `lbl_`. 
+Label names and label values should be compatible with a [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) specification.
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki.source.awsfirehose.md:69]:
```suggestion
You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels.
Label names must be prefixed with `lbl_`. 
Label names and label values must be compatible with the [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) specification.
```

**Comment by clayton-cornell** [comment]
Doc content looks OK to me. It's over to @grafana/grafana-alloy-maintainers for code review now :-)

File: docs/sources/reference/components/loki.source.awsfirehose.md [hunk 5]
```diff
@@ -65,6 +65,20 @@ The component will start an HTTP server on the configured port and address with
 
 - `/awsfirehose/api/v1/push` - accepting `POST` requests compatible with [AWS Firehose HTTP Specifications](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html).
 
+You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels.
+Label names must be prefixed with `lbl_`. 
+Label names and label values must be compatible with the [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) specification.
+
+Example of the valid `X-Amz-Firehose-Common-Attributes` value with two custom labels:
```
Comment (tpaschalis) [inline docs/sources/reference/components/loki.source.awsfirehose.md:72]: When I read this, it wasn't immediately apparent whether these labels are set on the component side or on the AWS firehose side and the component is just reading them out.

Can we make it more explicit and explain where these labels will be applied to (iiuc the read log lines)?

File: internal/component/loki/source/aws_firehose/internal/handler_test.go [hunk 6]
```diff
@@ -468,3 +468,277 @@ func keepLabelRule(src, dst string) *relabel.Config {
...
+		Assert func(t *testing.T, res *httptest.ResponseRecorder, entries []loki.Entry)
+
+		// AssertMetrics is an optional assertion over the collected metrics
+		AssertMetrics      func(t *testing.T, m []*dto.MetricFamily)
+		StaticLabelsConfig string
+	}
+
+	tests := map[string]testcase{
+		"direct put data, static labels": {
+			Body: readTestData(t, "testdata/direct_put.json"),
```
Comment (tpaschalis) [inline internal/component/loki/source/aws_firehose/internal/handler_test.go:490]: Do we need to commit this file?

Comment (andriikushch) [reply internal/component/loki/source/aws_firehose/internal/handler_test.go:490]: @tpaschalis, I think this file was already committed.

https://github.com/grafana/alloy/blob/main/internal/component/loki/source/aws_firehose/internal/testdata/direct_put.json

Comment (tpaschalis) [reply internal/component/loki/source/aws_firehose/internal/handler_test.go:490]: Ah, apologies then.

Comment (andriikushch) [reply internal/component/loki/source/aws_firehose/internal/handler_test.go:490]: No worries :)

**Review by tpaschalis — APPROVED** [review]
LGTM % the doc nits.

File: docs/sources/reference/components/loki.source.awsfirehose.md [hunk 7]
```diff
@@ -65,6 +65,20 @@ The component will start an HTTP server on the configured port and address with
 
 - `/awsfirehose/api/v1/push` - accepting `POST` requests compatible with [AWS Firehose HTTP Specifications](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html).
 
+You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels. The header can be configured in the **Parameters** section of the Amazon Data Firehose delivery stream configuration.
```
Comment (clayton-cornell) [inline docs/sources/reference/components/loki.source.awsfirehose.md:68]:
```suggestion
You can use the [X-Amz-Firehose-Common-Attributes](https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html) header to set extra static labels. 
You can configure the header in the **Parameters** section of the Amazon Data Firehose delivery stream configuration. 
```

**Review by clayton-cornell — APPROVED** [review]
Minor doc suggestion.

**Pull request merged**